### PR TITLE
Dusk impl

### PR DESCRIPTION
**Pull request by alichraghi**
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-const gpu = @import("gpu.zig");
 const builtin = @import("builtin");
+const gpu = @import("mach-gpu");
+const vulkan = @import("vulkan.zig");
 
-var didInit = false;
+var inited = false;
 var allocator: std.mem.Allocator = undefined;
 
-const Implementation = @import("vulkan/impl.zig");
```
Comment (Beyley) [inline src/main.zig]: This line was here to allow the code to more easily be shared between backends, where you would just reimplement the `Implementation` struct, and then all the other code inside `Interface` gets shared between d3d12, vk, mtl, etc.

File: libs/mach-gpu
Comment (Beyley) [inline libs/mach-gpu:1]: Are we required to depend on this through a submodule?

**Review by Beyley — CHANGES_REQUESTED** [review]

File: libs/mach-gpu
Comment (alichraghi) [reply libs/mach-gpu:1]: yes. see https://github.com/ziglang/zig/pull/15382

File: src/main.zig (hunk 1 above)
Comment (alichraghi) [reply src/main.zig]: yes. but we are not doing this yet as far as i can say

Comment (Beyley) [reply src/main.zig]: We aren't at the moment, but id rather do the pattern early on than try to bolt it on after the fact to get the codebase ready for the next backend (once vk is in a usable state and we are ready to start another one)

**Pull request closed without merging**